### PR TITLE
Enable nullability in FontDialog

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in FontDialog.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6870)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs [hunk 1]
```diff
@@ -155,7 +153,7 @@ public Font Font
         {
             get
             {
-                Font result = font;
+                Font? result = font;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs:156]:
```suggestion
                Font? result = font ?? Control.DefaultFont;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs [hunk 2]
```diff
@@ -155,11 +153,7 @@ public Font Font
         {
             get
             {
-                Font result = font;
-                if (result is null)
-                {
-                    result = Control.DefaultFont;
-                }
+                Font? result = font ?? Control.DefaultFont;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs:156]: `Control.DefaultFont` can't be null
https://github.com/dotnet/winforms/blob/09120ef4871dccfa3b4c09e33ad96648cf3f4c64/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs#L1811-L1823
```suggestion
                Font result = font ?? Control.DefaultFont;
```

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]: `SystemFonts.MessageBoxFont` is nullable though. I guess when we enable nullability in Control we could use the null-forgiving operator there.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs]: Ah, right. All good then

File: src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs [hunk 3]
```diff
@@ -155,11 +153,7 @@ public Font Font
         {
             get
             {
-                Font result = font;
-                if (result is null)
-                {
-                    result = Control.DefaultFont;
-                }
+                Font result = font ?? Control.DefaultFont;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/FontDialog.cs:156]:
```suggestion
                Font? result = font ?? Control.DefaultFont;
```

**Review by RussKie — APPROVED** [review]

**Pull request merged**